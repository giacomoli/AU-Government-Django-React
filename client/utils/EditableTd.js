import React from 'react'
import PropTypes from 'prop-types'

class EditableTd extends React.Component {
  static propTypes = {
    fieldName: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    children: PropTypes.number.isRequired,
  }
  constructor(props) {
    super(props)

    this.state = {
      flip: false,
      value: props.children,
    }
  }
  handleOnClick = (e) => {
    this.stopPropagation(e)
    this.setState(state => ({ flip: !state.flip }))
  }
  handleOnBlur = () => {
    this.setState((state) => {
      if (state.value || state.value === 0) {
        return { flip: false }
      }

      return null
    })
  }
  handleOnChange = (event) => {
    const { value } = event.target

    if (!isNaN(value)) {
      this.setState({ value })
      this.props.onChange(this.props.id, value)
    } else {
      this.props.onChange(this.props.id, this.props.children)
    }
  }
  stopPropagation = e => e.stopPropagation()
  render() {
    /* eslint-disable */
    return (
      <td onClick={this.handleOnClick}>
        {this.state.flip ? (
          <input
            onChange={this.handleOnChange}
            onClick={this.stopPropagation}
            onBlur={this.handleOnBlur}
            name={this.props.fieldName}
            value={this.state.value}
            type="text"
          />
        ) : (
          this.props.children
        )}
      </td>
    )
  }
}

export default EditableTd
