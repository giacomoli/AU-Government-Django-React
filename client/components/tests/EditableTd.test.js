import React from 'react'
import renderer from 'react-test-renderer'
import EditableTd from '../../utils/EditableTd'

/* eslint-disable */
describe('EditableTd', () => {

  it('should render EditableTd with props', () => {
    const props = {
      fieldName: 'Test text',
      onChange: jest.fn(),
      children: 1,
    }
    const wrapper = renderer.create(
        <EditableTd
          fieldName={props.fieldName}
          onChange={props.onChange}
          children={props.children}
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
