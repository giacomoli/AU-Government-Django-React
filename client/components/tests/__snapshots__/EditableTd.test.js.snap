// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTd should render EditableTd with props 1`] = `
<td
  onClick={[Function]}
>
  1
</td>
`;
